cardano-db-sync/src/Cardano/DbSync/Default.hs

```diff
@@ -82,17 +88,55 @@ applyAndInsertBlockMaybe syncEnv cblk = do
         rollbackFromBlockNo syncEnv (blockNo cblk)
         insertBlock syncEnv cblk applyRes True tookSnapshot
         liftIO $ setConsistentLevel syncEnv Consistent
+        -- now that we have caught up with the tip of the chain
+        -- we can put the constraints on Reward + EpochStake tables
+        -- as a speed improvement.
+        lift addRewardTableConstraint
```

Comment:
I think there should be an `unless` check in case the unique key exists already. Otherwise there would be `constraind already exists errors`

Reply:
ah ok, maybe I do that check inside the function it self. Actually your'e right would be more obvious if that check happens here 👍